File: src/main/java/dev/lampirg/consultationappointment/service/notification/SimpleEmailService.java
package dev.lampirg.consultationappointment.service.notification;

import org.springframework.context.annotation.Profile;
import org.springframework.mail.MailSender;
import org.springframework.mail.SimpleMailMessage;
import org.springframework.stereotype.Service;

@Service
@Profile("email & prod")
public class SimpleEmailService implements EmailService<SimpleMailMessage> {

    private MailSender mailSender;

    public SimpleEmailService(MailSender mailSender) {
        this.mailSender = mailSender;
    }

    @Override
    public void sendEmail(SimpleMailMessage message) {
        mailSender.send(message);
    }
}

File: src/main/kotlin/dev/lampirg/consultationappointment/service/notification/SimpleEmailService.kt
package dev.lampirg.consultationappointment.service.notification

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Profile
import org.springframework.mail.MailSender
import org.springframework.mail.SimpleMailMessage
import org.springframework.stereotype.Service

@Service
@Profile("email & prod")
class SimpleEmailService @Autowired constructor(private val mailSender: MailSender) : EmailService<SimpleMailMessage> {
    override fun sendEmail(message: SimpleMailMessage) {
        mailSender.send(message)
    }
}